internal/otkpart/partition.go

```diff
@@ -0,0 +1,28 @@
+package otkpart
+
+import (
+	"github.com/osbuild/images/pkg/disk"
+)
+
+type PartitionTable struct {
+	Const Const `json:"const"`
+}
+
+type Const struct {
+	KernelOptsList []string `json:"kernel_opts_list"`
+	// we generate this for convenience for otk users, so that they
+	// can write, e.g. "filesystem.partition_map.boot.uuid"
+	PartitionMap map[string]Partition `json:"partition_map"`
+	Internal     Internal             `json:"internal"`
+}
+
+// "exported" view of partitions, this is an API so only add things here
+// that are really needed and unlikely to change
+type Partition struct {
```

Comment:
I think some linter somewhere might complain because conventionally docstrings of exported types and functions should start with the name of the object they're documenting, but we'll fix that in post.